lightning_client.go

```diff
@@ -4132,6 +4139,20 @@ func (s *lightningClient) SendCustomMessage(ctx context.Context,
 	return err
 }
 
+// SignMessage signs a message with this node's private key.
+// The returned signature string is zbase32 encoded and pubkey
+// recoverable, meaning that only the message digest and signature
+// are needed for verification.
+func (s *lightningClient) SignMessage(ctx context.Context,
+	data []byte) (string, error) {
+	rpcCtx := s.adminMac.WithMacaroonAuth(ctx)
+
+	rpcReq := &lnrpc.SignMessageRequest{Msg: data}
+	rpcRes, err := s.client.SignMessage(rpcCtx, rpcReq)
```

Comment:
Need to add an `if err != nil ` check here. Otherwise `rpcRes.Signature` below will cause a nil pointer panic.

Reply:
Ah, nice catch. Should have caught that one.